addons/sale_loyalty/models/sale_order.py

```diff
@@ -165,15 +165,23 @@ def _discountable_order(self, reward):
             # Ignore lines from this reward
             if not line.product_uom_qty or not line.price_unit:
                 continue
-            tax_data = line._convert_to_tax_base_line_dict()
-            # To compute the discountable amount we get the fixed tax amount and
-            # subtract it from the order total. This way fixed taxes will not be discounted
-            tax_data['taxes'] = tax_data['taxes'].filtered(lambda t: t.amount_type == 'fixed')
-            tax_results = self.env['account.tax']._compute_taxes([tax_data])
-            totals = list(tax_results['totals'].values())[0]
-            discountable += line.price_total - totals['amount_tax']
+            tax_data = line.tax_id.compute_all(
+                line.price_unit,
+                quantity=line.product_uom_qty,
+                product=line.product_id,
+                partner=line.order_partner_id,
+            )
+            # To compute the discountable amount we get the subtotal and add
+            # non-fixed tax totals. This way fixed taxes will not be discounted
             taxes = line.tax_id.filtered(lambda t: t.amount_type != 'fixed')
-            discountable_per_tax[taxes] += totals['amount_untaxed']
+            discountable += tax_data['total_excluded'] + sum(
+                tax['amount'] for tax in tax_data['taxes'] if tax['id'] in taxes.ids
+            )
+            line_price = line.price_unit * line.product_uom_qty * (1 - line.discount / 100)
```

Comment:
```suggestion
            line_price = line.price_unit * line.product_uom_qty * (1 - (line.discount or 0.0) / 100)
```
last little change to be on the safe side, like it's done in `sale`.

Reply:
Sure, added the change for consistency, though the ORM already ensures `Float` fields always return 0.0 as the falsy value:
https://github.com/odoo/odoo/blob/990aae55b1a2b95765a848b6fc122bb69f768ce1/odoo/fields.py#L1538-L1550